### ISSUE TITLE
optional param in schema cause error : Cannot set property 'message' of undefined (v4.1)

### ISSUE THREAD

**Issue by vitaliybilous** [ctavan/express-validator#398]
**lib : 4.1,**
**node :  8.1.4,**
**os : ubuntu 14.04**

Use legacy API (schema) for validation.
My code:
```javascript
export const validatePostPageData = async (req, res, next) => {
  /*
  POST data : { data : { props : "ABC" } }
  */
  req.checkBody({
    'data.props' : {
      optional: true
    }
  });

  const result = await req.getValidationResult();

  if (!result.isEmpty()) {
    res.status(400).send({error: result.array().map(({param, msg}) => ({param, msg}))});
    return;
  }

  next();
};
```
Displayed error in console:
`
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot set property 'message' of undefined
`

If remove the optional rule and add other - all work fine

**Comment by gustavohenke** [comment]
Thanks for the report. This is indeed a bug when using schema validation + `optional` as the first found key in a parameter object.

**Comment by gustavohenke** [comment]
Fix published in v4.1.1.
